x-pack/test/api_integration/apis/ml/data_visualizer/get_field_stats.ts

```diff
@@ -0,0 +1,238 @@
+        { fieldName: 'airline', type: 'keyword', cardinality: 19 },
+        { fieldName: 'type', type: 'text', cardinality: 0 },
+        { fieldName: 'type.keyword', type: 'keyword', cardinality: 1 },
+      ],
+      samplerShardSize: -1, // No sampling, as otherwise counts would vary on each run.
+      timeFieldName: '@timestamp',
+      earliest: 1454889600000, // February 8, 2016 12:00:00 AM GMT
+      latest: 1454976000000, // February 9, 2016 12:00:00 AM GMT
+      maxExamples: 10,
+    },
+    expected: {
+      responseCode: 200,
+      responseBody: [
+        { fieldName: '@timestamp', count: 1733, earliest: 1454889602000, latest: 1454975948000 },
+        {
+          fieldName: '@version.keyword',
+          isTopValuesSampled: false,
+          topValues: [{ key: '1', doc_count: 1733 }],
+          topValuesSampleSize: 1733,
+          topValuesSamplerShardSize: -1,
+        },
+        {
+          fieldName: 'airline',
+          isTopValuesSampled: false,
+          topValues: [{ key: 'AAL', doc_count: 1733 }],
+          topValuesSampleSize: 1733,
+          topValuesSamplerShardSize: -1,
+        },
+        {
+          fieldName: 'type.keyword',
+          isTopValuesSampled: false,
+          topValues: [{ key: 'farequote', doc_count: 1733 }],
+          topValuesSampleSize: 1733,
+          topValuesSamplerShardSize: -1,
+        },
+        { fieldName: 'type', examples: ['farequote'] },
+      ],
+    },
+  };
+
+  const errorTestData = {
+    testTitle: 'returns error for index which does not exist',
+    index: 'ft_farequote_not_exists',
+    user: USER.ML_POWERUSER,
+    requestBody: {
+      query: { bool: { must: [{ match_all: {} }] } },
+      fields: [
+        { type: 'number', cardinality: 0 },
+        { fieldName: 'responsetime', type: 'number', cardinality: 4249 },
+      ],
+      samplerShardSize: 1000, // No sampling, as otherwise counts could vary on each run.
+      timeFieldName: '@timestamp',
+      maxExamples: 10,
+    },
+    expected: {
+      responseCode: 404,
+      responseBody: {
+        statusCode: 404,
+        error: 'Not Found',
+        message:
+          '[index_not_found_exception] no such index [ft_farequote_not_exists], with { resource.type="index_or_alias" & resource.id="ft_farequote_not_exists" & index_uuid="_na_" & index="ft_farequote_not_exists" }',
+      },
+    },
+  };
+
+  function compareByFieldName(a: { fieldName: string }, b: { fieldName: string }) {
+    if (a.fieldName < b.fieldName) {
+      return -1;
+    }
+    if (a.fieldName > b.fieldName) {
+      return 1;
+    }
+    return 0;
+  }
+
+  describe('get_field_stats', function() {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    it(`${metricFieldsTestData.testTitle}`, async () => {
+      const { body } = await supertest
+        .post(`/api/ml/data_visualizer/get_field_stats/${metricFieldsTestData.index}`)
```

Comment:
This request code exists three times in this file. Might be worth to define this in a separate function, something like
```
  async function runGetFieldStatsRequest(
    index: string,
    user: USER,
    requestBody: object,
    expectedResponsecode: number
  ): Promise<any> {
    const { body } = await supertest
      .post(`/api/ml/data_visualizer/get_field_stats/${index}`)
      .auth(user, ml.securityCommon.getPasswordForUser(user))
      .set(COMMON_HEADERS)
      .send(requestBody)
      .expect(expectedResponsecode);

    return body;
  }
```
Alternatively, this test could be converted to use a `testDataList`, which would help with future plans to align the structure of test files.

Reply:
Done in https://github.com/elastic/kibana/pull/64165/commits/2b63f4504c03ec5d78ccef7cba518a1511be3b15. Note I haven't used a `testDataList` here as the checks for each test have some differences.